File: web/src/api/jxc/stock.js
import service from '@/utils/request'

// 库存中心
export const getStockPage = (params) => service.get('/jxc/stock/page', { params })
export const getStockLogPage = (params) => service.get('/jxc/stock/log/page', { params })
export const directIn = (data) => service.post('/jxc/stock/direct-in', data)
export const directOut = (data) => service.post('/jxc/stock/direct-out', data)

File: server/router/jxc/stock.go
package jxc

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type StockRouter struct{}

func (r *StockRouter) InitStockRouter(Router *gin.RouterGroup) {
	recordGroup := Router.Group("jxc").Use(middleware.OperationRecord())
	readOnlyGroup := Router.Group("jxc")

	// ===== 库存中心 =====
	readOnlyGroup.GET("stock/page", stockApi.GetStockPage)
	readOnlyGroup.GET("stock/log/page", stockApi.GetStockLogPage)
	recordGroup.POST("stock/direct-in", stockApi.DirectIn)
	recordGroup.POST("stock/direct-out", stockApi.DirectOut)
}
